File: 10322.cpp
#include <iostream>
#include <string>
#include <queue>
using namespace std;

string inp;
void input() {
    cin >> inp;
}
bool compute() {
    int cnt_w = 0, cnt_o = 0, cnt_l = 0, cnt_f = 0;
    queue<char> qWord;
    for (char c : inp)
        qWord.push(c);
    while (!qWord.empty()) {
        char c = qWord.front();
        qWord.pop();
        switch (c) {
            case 'w':
                cnt_w += 1;
                if (cnt_o | cnt_l | cnt_f != 0)
                    return false;
                break;

            case 'o':
                cnt_o += 1;
                if (cnt_w == 0)
                    return false;
                break;

            case 'l':
                cnt_l += 1;
                if (cnt_w == 0 || cnt_o == 0 || cnt_w != cnt_o)
                    return false;
                break;

            case 'f':
                cnt_f += 1;
                if (cnt_w == 0 || cnt_o == 0 || cnt_l == 0 || !(cnt_w == cnt_o && cnt_o == cnt_l))
                    return false;
                break;
        }
        if (cnt_w == cnt_o && cnt_o == cnt_l && cnt_l == cnt_f)
            cnt_w = cnt_o = cnt_l = cnt_f = 0;
    }
    if (cnt_w == cnt_o && cnt_o == cnt_l && cnt_l == cnt_f)
        return true;
    else
        return false;
}
void output(bool res) {
    cout << res;
}
int main() {
    input();
    output(compute());
    return 0;
}

File: 10322.py
import sys
from collections import deque


input = sys.stdin.readline


def solution() -> bool:
    cnt_w, cnt_o, cnt_l, cnt_f = 0, 0, 0, 0
    while len(word):
        c = word.popleft()
        locals()['cnt_' + c] += 1
        if c == 'w':
            cnt_w += 1
            if cnt_o | cnt_l | cnt_f != 0:
                return False
        elif c == 'o':
            cnt_o += 1
            if cnt_w <= 0:
                return False
        elif c == 'l':
            cnt_l += 1
            if cnt_w == 0 or cnt_o == 0 or cnt_w != cnt_o:
                return False
        elif c == 'f':
            cnt_f += 1
            if cnt_w == 0 or cnt_o == 0 or cnt_w == 0 or not (cnt_w == cnt_o == cnt_l):
                return False

        if cnt_w == cnt_o == cnt_l == cnt_f:
            cnt_w = cnt_o = cnt_l = cnt_f = 0

    if cnt_w == cnt_o == cnt_l == cnt_f:
        return True
    else:
        return False


if __name__ == '__main__':
    word = deque(list(input().rstrip()))
    sol = solution()
    print(int(sol))
